src/Symfony/Component/DependencyInjection/Argument/TaggedIteratorArgument.php

```diff
@@ -24,16 +24,18 @@ class TaggedIteratorArgument extends IteratorArgument
     private ?string $defaultPriorityMethod;
     private bool $needsIndexes;
     private array $exclude;
+    private bool $autoExcludeReferencingService = true;
 
     /**
-     * @param string      $tag                   The name of the tag identifying the target services
-     * @param string|null $indexAttribute        The name of the attribute that defines the key referencing each service in the tagged collection
-     * @param string|null $defaultIndexMethod    The static method that should be called to get each service's key when their tag doesn't define the previous attribute
-     * @param bool        $needsIndexes          Whether indexes are required and should be generated when computing the map
-     * @param string|null $defaultPriorityMethod The static method that should be called to get each service's priority when their tag doesn't define the "priority" attribute
-     * @param array       $exclude               Services to exclude from the iterator
+     * @param string      $tag                           The name of the tag identifying the target services
+     * @param string|null $indexAttribute                The name of the attribute that defines the key referencing each service in the tagged collection
+     * @param string|null $defaultIndexMethod            The static method that should be called to get each service's key when their tag doesn't define the previous attribute
+     * @param bool        $needsIndexes                  Whether indexes are required and should be generated when computing the map
+     * @param string|null $defaultPriorityMethod         The static method that should be called to get each service's priority when their tag doesn't define the "priority" attribute
+     * @param array       $exclude                       Services to exclude from the iterator
+     * @param array       $autoExcludeReferencingService Whether to automatically exclude the referencing service from the iterator
      */
-    public function __construct(string $tag, string $indexAttribute = null, string $defaultIndexMethod = null, bool $needsIndexes = false, string $defaultPriorityMethod = null, array $exclude = [])
+    public function __construct(string $tag, string $indexAttribute = null, string $defaultIndexMethod = null, bool $needsIndexes = false, string $defaultPriorityMethod = null, array $exclude = [], bool $autoExcludeReferencingService = true)
```

Comment:
excludeSelf?

Reply:
I started with it and found disturbing to talk about `self` here as it literally means `TaggedIteraotrArgument::class`. I guess we don't care: attributes-specfic. Renamed